lib/SILOptimizer/Transforms/CSE.cpp

```diff
@@ -267,21 +301,29 @@ class HashVisitor : public SILInstructionVisitor<HashVisitor, llvm::hash_code> {
   }
 
   hash_code visitApplyInst(ApplyInst *X) {
-    OperandValueArrayRef Operands(X->getAllOperands());
-    return llvm::hash_combine(X->getKind(), X->getCallee(),
-                              llvm::hash_combine_range(Operands.begin(),
-                                                       Operands.end()),
-                              X->hasSubstitutions());
+    auto Operands = X->getOperandValues();
+    SmallVector<SILValue, 4> OperandValues;
+    llvm::transform(
+        Operands, std::back_inserter(OperandValues),
```

Comment:
One thought. Is it possible to create a range that does this transform and pass that to hash_combine instead? That way we do not intermediate arrays (and potential mallocs).

(Keep in mind I am just tagging this one, but I am talking about all of the places that this was done.

Reply:
I'll try to do this

Reply:
I added a transform function as an optional argument to SILInstruction::getOperandValues. It may be useful in other code as well.